docs/content/docs/dev/table/functions/ptfs.md

```diff
@@ -0,0 +1,1349 @@
+-------------------
+
+The following examples demonstrate how a PTF can accept and transform tables. The `@ArgumentHint` specifies that the function
+accepts a table as an argument, rather than just a scalar row value. In both examples, the `eval()` method is invoked for
+each row in the input table. Additionally, the `@ArgumentHint` not only indicates that the function can process a table,
+but also defines how the function interprets the table - whether using row or set semantics.
+
+### Greeting
+
+An example that adds a greeting to each incoming customer.
+
+{{< tabs "0137eeed-3d13-455c-8e2f-5e164da9f844" >}}
+{{< tab "Java" >}}
+```java
+import org.apache.flink.table.annotation.*;
+import org.apache.flink.table.api.*;
+import org.apache.flink.table.functions.ProcessTableFunction;
+import org.apache.flink.types.Row;
+import static org.apache.flink.table.api.Expressions.*;
+
+// A PTF that takes a table argument, conceptually viewing the table as a row.
+// The result is never stateful and derived purely based on the current row.
+public static class Greeting extends ProcessTableFunction<String> {
+    public void eval(@ArgumentHint(ArgumentTrait.TABLE_AS_ROW) Row input) {
+        collect("Hello " + input.getFieldAs("name") + "!");
+    }
+}
+
+TableEnvironment env = TableEnvironment.create(...);
+
+// Call the PTF with row semantics "inline" (without registration) in Table API
+env.fromValues("Bob", "Alice", "Bob")
+  .as("name")
+  .process(Greeting.class)
+  .execute()
+  .print();
+
+// For SQL, register the PTF upfront
+env.executeSql("CREATE VIEW Names(name) AS VALUES ('Bob'), ('Alice'), ('Bob')");
+env.createFunction("Greeting", Greeting.class);
+
+// Call the PTF with row semantics in SQL
+env.executeSql("SELECT * FROM Greeting(TABLE Names)").print();
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+The results of both Table API and SQL look similar to:
+
+```text
++----+--------------------------------+
+| op |                         EXPR$0 |
++----+--------------------------------+
+| +I |                     Hello Bob! |
+| +I |                   Hello Alice! |
+| +I |                     Hello Bob! |
++----+--------------------------------+
+```
+
+### Greeting with Memory
+
+An example that adds a greeting for each incoming customer, taking into account whether the customer has been greeted
+previously.
+
+{{< tabs "0237eeed-3d13-455c-8e2f-5e164da9f844" >}}
+{{< tab "Java" >}}
+```java
+import org.apache.flink.table.annotation.*;
+import org.apache.flink.table.api.*;
+import org.apache.flink.table.functions.ProcessTableFunction;
+import org.apache.flink.types.Row;
+import static org.apache.flink.table.api.Expressions.*;
+
+// A PTF that takes a table argument, conceptually viewing the table as a set.
+// The result can be stateful and derived based on the current row and/or
+// previous rows in the set.
+// The call's partitioning defines the size of the set.
+public static class GreetingWithMemory extends ProcessTableFunction<String> {
```

Comment:
how about adding example into Flink-examples module to make sure they are always compilable. I remember we had a number of issues when examples from doc couldn't be compiled

Reply:
Sure, we can do this as a follow up. Nevertheless people also love examples in docs just to get a feeling how it all fits together.